### PR TITLE
Add missed files from MRI 2.3.7

### PR DESCRIPTION
**Pull request by eregon**

### CODE REVIEW

**Comment by nirvdrum** [comment]
It looks fine to me, but I'll let Chris review since he did the original work.

File: test/mri/tests/ruby/test_vm_dump.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  end
+
+  def test_darwin_segv_in_syscall
+    assert_darwin_vm_dump_works('-e1.times{Process.kill :SEGV,$$}')
+  end
+
+  def test_darwin_invalid_access
+    assert_darwin_vm_dump_works(['-rfiddle', '-eFiddle.dlunwrap(100).class'])
+  end
+end
```
Comment (chrisseaton) [inline test/mri/tests/ruby/test_vm_dump.rb]: I think we had difficulty excluding these tests for some reason in the past - you haven't excluded them at all here and they aren't going to work are they?

File: test/mri/tests/cext-ruby/symbol/noninterned_name.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+    def noninterned_name(prefix = "")
+      prefix += "_#{Thread.current.object_id.to_s(36).tr('-', '_')}"
+      begin
+        name = "#{prefix}_#{rand(0x1000).to_s(16)}_#{Time.now.usec}"
+      end while Bug::Symbol.find(name)
+      name
+    end
+  end
+end
```
Comment (chrisseaton) [inline test/mri/tests/cext-ruby/symbol/noninterned_name.rb]: How did this one get left behind?

Comment (eregon) [reply test/mri/tests/cext-ruby/symbol/noninterned_name.rb]: No idea, I just noticed it wasn't added after redoing the 2.3.7 import.

**Comment by chrisseaton** [comment]
With our current system of the the unmodified commit it's basically impossible to rebase this and understand what's going on. I'll make a note to update these files because I need to do another clean commit anyway after updating Psych. We need a better system for the reference commit really.

**Comment by eregon** [comment]
Sounds good, I extracted the "Ignore all .dist directories at the root" commit in #1319.

**Pull request closed without merging**